### DISCUSSION TITLE
How to customize onCatch CatchBoundary handler?

### DISCUSSION THREAD

**Discussion by bpinto** [TanStack/router#1488] [Q&A] [answered]
It's possible to configure a custom error component per route (or a default error component) but how can I customise the `onCatch` function? My use case is for notifying (sending a request) to my error reporting tool whenever an error occurs.

**Comment by tannerlinsley** [comment, marked as answer]
You can get all errors for a route via the onError option which takes a function that receives the error. You must rethrow the error if you want TanStack Router to keep handling it!

**Reply by bpinto** [reply]
Am I right that this `onError` is only triggered for errors during a navigation or preload event and not for an error during rendering lifecycle?

If so, would it make sense to add a `defaultErrorBoundary` option to router so that we could implement a custom `onCatch` function at the error boundary level which would then be called on any situation? During navigation or rendering lifecycle.

Update: Maybe that is not a good idea. But I'm not quite sure how to have `onError` called for rendering lifecycle errors.

**Comment by bpinto** [comment]
Thank you! I somehow missed this option.
